core/testcontainers/core/docker_client.py

```diff
@@ -98,36 +98,36 @@ def find_host_network(self) -> Optional[str]:
                         except ipaddress.AddressValueError:
                             continue
                         if docker_host in subnet:
-                            return network.name
+                            return cast(str, network.name)
         except ipaddress.AddressValueError:
             pass
         return None
 
-    def port(self, container_id: str, port: int) -> int:
+    def port(self, container_id: str, port: int) -> str:
         """
         Lookup the public-facing port that is NAT-ed to :code:`port`.
         """
         port_mappings = self.client.api.port(container_id, port)
         if not port_mappings:
             raise ConnectionError(f"Port mapping for container {container_id} and port {port} is " "not available")
-        return port_mappings[0]["HostPort"]
+        return cast(str, port_mappings[0]["HostPort"])
 
-    def get_container(self, container_id: str) -> Container:
+    def get_container(self, container_id: str) -> dict[str, Any]:
```

Comment:
This looks like a regression to me.

(I would argue that nothing should return a dict directly, ever, but always a typed dictionary, dataclass or something similar with described fields. In my experience, interfaces with dict always end in maintenance headache)

Reply:
in the opinion of the project, it is also a maintenance burden to return types from other libraries. either way, I think breaking the actual return and changing runtime behavior would be worse than fixing the type.